src/semantics.lua

```diff
@@ -1859,25 +1863,111 @@ local function semanticsSmol(sources, main)
 						purpose = "do-statement expression",
 						expectedCount = 1,
 						givenCount = #out,
-						location = pExpression.location,
+						location = pStatement.location,
 					}
 				elseif not areTypesEqual(out[1].type, UNIT_TYPE) then
 					Report.TYPES_DONT_MATCH {
 						purpose = "do-statement expression",
 						expectedType = "Unit",
-						expectedLocation = pExpression.expression.location,
+						expectedLocation = pStatement.expression.location,
 						givenType = showType(out[1].type),
-						location = pExpression.expression.location,
+						location = pStatement.expression.location,
 					}
 				end
 
 				return evaluation
+			elseif pStatement.tag == "if-statement" then
+				local conditionEvaluation, conditionOut = compileExpression(pStatement.condition, scope)
+				assert(#conditionOut ~= 0)
+				if #conditionOut > 1 then
+					Report.WRONG_VALUE_COUNT {
+						purpose = "if-statement condition",
+						expectedCount = 1,
+						givenCount = #conditionOut,
+						location = pStatement.location,
+					}
+				elseif not areTypesEqual(conditionOut[1].type, BOOLEAN_TYPE) then
+					Report.TYPES_DONT_MATCH {
+						purpose = "if-statement condition",
+						expectedType = "Boolean",
+						expectedLocation = pStatement.condition.location,
+						givenType = showType(conditionOut[1].type),
+						location = pStatement.condition.location,
+					}
+				end
+
+				local unclear = function(a, b)
```

Comment:
NIT: `local function unclear(a, b)` is probably better
An `assertis(a, "maybe")` and a comment stating return type / purpose should be used.
It should also be lifted out of here, in case it's useful elsewhere.